homeassistant/helpers/trigger_template_entity.py

```diff
@@ -176,18 +177,41 @@ def restore_attributes(self, last_state: State) -> None:
                 extra_state_attributes[attr] = last_state.attributes[attr]
             self._rendered[CONF_ATTRIBUTES] = extra_state_attributes
 
+    def _render_availability_template(self, variables: dict[str, Any]) -> None:
+        """Render availability template."""
+        self.error = False
+        rendered = {**self._static_rendered, **self._rendered}
+        key = CONF_AVAILABILITY
+        try:
+            if key in self._to_render_simple:
+                rendered[key] = self._config[key].async_render(
+                    variables,
+                    parse_result=key in self._parse_result,
+                )
+        except TemplateError as err:
+            logging.getLogger(f"{__package__}.{self.entity_id.split('.')[0]}").error(
+                "Error rendering %s template for %s: %s", key, self.entity_id, err
+            )
+            self.error = True
+        self._rendered = rendered
+
     def _render_templates(self, variables: dict[str, Any]) -> None:
         """Render templates."""
+        rendered = dict(self._rendered)
+        if CONF_AVAILABILITY in rendered and rendered[CONF_AVAILABILITY] is False:
+            return
         try:
-            rendered = dict(self._static_rendered)
-
             for key in self._to_render_simple:
+                if key == CONF_AVAILABILITY:
+                    continue
                 rendered[key] = self._config[key].async_render(
                     variables,
                     parse_result=key in self._parse_result,
                 )
 
             for key in self._to_render_complex:
+                if key == CONF_AVAILABILITY:
+                    continue
```

Comment:
Can we add a test which exercises this path?

Reply:
As with below comment, `availability` can not be complex, so removing this check.